guides/v2.3/graphql/queries/products-234.md

```diff
@@ -0,0 +1,1144 @@
+---
+group: graphql
+title: products query (Magento 2.3.4 and later)
+---
+
+The `products` query allows you to search for catalog items.
+
+## Syntax
+
+```graphql
+products(
+  search: String
+  filter: ProductAttributeFilterInput
+  pageSize: Int
+  currentPage: Int
+  sort: ProductSortFilterInput
+): Products
+```
+
+## Input attributes
+
+Each query attribute is defined below:
+
+Attribute |  Data type | Description
+--- | --- | ---
+`search` | String | Performs a full-text search using the specified key words. This attribute is optional. S
+`filter` | ProductAttributeFilterInput | Identifies which attributes to search for and return. This attribute is required. See [filter attribute](#filter) object for more information.
```

Comment:
If this attribute is required why isn't it denoted with `ProductAttributeFilterInput!` in the datatype column?